### PR TITLE
#511 Add AppCategory entity

### PR DESCRIPTION
**Pull request by jo-elimuai**

### CODE REVIEW

**Comment by jo-elimuai** [comment]
screen shot 2017-09-16 at 18 05 19

File: src/main/java/ai/elimu/dao/jpa/AppCategoryDaoJpa.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package ai.elimu.dao.jpa;
+
+import ai.elimu.dao.AppCategoryDao;
+import ai.elimu.model.project.AppCategory;
+
+public class AppCategoryDaoJpa extends GenericDaoJpa<AppCategory> implements AppCategoryDao {
+
```
Comment (sladomic) [inline src/main/java/ai/elimu/dao/jpa/AppCategoryDaoJpa.java]: Why is this class empty?

File: src/main/webapp/WEB-INF/i18n/messages_ar.properties [hunk 2]
```diff
@@ -24,6 +24,7 @@ application.content=محتوى التطبيق
 application.opened.events=Application opened events
 application.versions=Application versions
 applications=Applications
+app.categories=App categories
```
Comment (sladomic) [inline src/main/webapp/WEB-INF/i18n/messages_ar.properties]: Doesn't look Arabic to me :)

File: src/main/webapp/WEB-INF/i18n/messages_ar.properties [hunk 3]
```diff
@@ -49,6 +50,7 @@ cover.image=Cover image
 creator=Creator
 crowdsourced=تعهيد جماعي
 curriculum=منهاج
+custom.projects=Custom projects
```
Comment (sladomic) [inline src/main/webapp/WEB-INF/i18n/messages_ar.properties]: Doesn't look Arabic to me :)

File: src/main/webapp/WEB-INF/i18n/messages_sw.properties [hunk 4]
```diff
@@ -24,6 +24,7 @@ application.content=Maudhui ya programu tumizi
 application.opened.events=Matukio yaliyofunguliwa ya programu tumizi
 application.versions=Matoleo ya programu tumizi
 applications=Programu tumizi
+app.categories=App categories
```
Comment (sladomic) [inline src/main/webapp/WEB-INF/i18n/messages_sw.properties]: Doesn't look Swahili to me :)

File: src/main/webapp/WEB-INF/i18n/messages_sw.properties [hunk 5]
```diff
@@ -49,6 +50,7 @@ cover.image=Picha kuu
 creator=Muundaji
 crowdsourced=Chanzo ni umma
 curriculum=Mitaala
+custom.projects=Custom projects
```
Comment (sladomic) [inline src/main/webapp/WEB-INF/i18n/messages_sw.properties]: Doesn't look Swahili to me :)

File: src/main/webapp/WEB-INF/jsp/project/app-category/list.jsp [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<content:title>
+    <fmt:message key="app.categories" /> (${fn:length(project.appCategories)})
+</content:title>
+
+<content:section cssId="appCategoryListPage">
+    <div class="section row">
+        ...
```
Comment (sladomic) [inline src/main/webapp/WEB-INF/jsp/project/app-category/list.jsp]: No yet finished?

File: src/main/java/ai/elimu/dao/jpa/AppCategoryDaoJpa.java (hunk 1 above)
Comment (jo-elimuai) [reply src/main/java/ai/elimu/dao/jpa/AppCategoryDaoJpa.java]: @sladomic It inherits basic functionality (create, read, update, delete) from a parent class. But we will probably soon add methods to it as needed.

File: src/main/webapp/WEB-INF/i18n/messages_ar.properties (hunk 2 above)
Comment (jo-elimuai) [reply src/main/webapp/WEB-INF/i18n/messages_ar.properties]: @sladomic Will have a native speaker go through and translate these later ;-)

File: src/main/webapp/WEB-INF/jsp/project/app-category/list.jsp (hunk 6 above)
Comment (jo-elimuai) [reply src/main/webapp/WEB-INF/jsp/project/app-category/list.jsp]: Work in progress.

**Comment by jo-elimuai** [comment]
screen shot 2017-09-16 at 18 44 09

**Pull request merged**